probes/hasLicenseFile/impl.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package hasLicenseFile
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "hasLicenseFile"
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	var outcome finding.Outcome
+	var msg string
+
+	if raw.LicenseResults.LicenseFiles == nil || len(raw.LicenseResults.LicenseFiles) == 0 {
+		outcome = finding.OutcomeNegative
+		msg = "project does not have a license file"
+	} else {
+		outcome = finding.OutcomePositive
+		msg = "project has a license file"
+	}
+
+	f, err := finding.NewWith(fs, Probe,
+		msg, nil,
+		outcome)
```

Comment:
we should include the location?

Reply:
Added in https://github.com/ossf/scorecard/pull/3465/commits/4827137bd5fd9fcd37febb3fe3fe6b7ca87c19da

Reply:
Not quite what I meant. The findings support having the location information as part of the struct, I wasn't referring to logging it.